app/src/main/java/com/example/tudeeapp/domain/exception/TudeeException.kt

```diff
@@ -1,8 +1,5 @@
 package com.example.tudeeapp.domain.exception
 
 abstract class TudeeException(message: String) : Exception(message)
-open class TaskException(message: String) : TudeeException(message)
-open class CategoryException(message: String="") : TudeeException(message)
-// customize your Exception
```

Comment:
In my opinion, I think the clean approach  is to make 2 exception:
Ex:
TasknotfoundException
&
Task database/error exception ->for the other operations of tasks

Reply:
No no, I didn't mean that
I mean you have a type of exception refer to any problem happened to dB, like add  failure , delete, and update
Then you will show to the user a generic message for that